libs/core/src/main/java/org/opensearch/core/tasks/TaskId.java

```diff
@@ -116,6 +142,11 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeLong(id);
     }
 
+    @Override
+    public void writeTo(OutputStream out) throws IOException {
+        out.write(this.taskIdProto.toByteArray());
```

Comment:
May want to consider special-casing the empty-node-id = 0 case here if you implement it per my earlier comments.